e2e_playwright/main_menu_test.py

```diff
@@ -22,3 +23,85 @@ def test_main_menu_images(themed_app: Page, assert_snapshot: ImageCompareFunctio
 
     element = themed_app.get_by_test_id("stMainMenuPopover")
     assert_snapshot(element, name="main_menu")
+
+
+def test_renders_settings_dialog_properly(
+    themed_app: Page, assert_snapshot: ImageCompareFunction
+):
+    themed_app.get_by_test_id("stMainMenu").click()
+
+    themed_app.get_by_text("Settings").click()
+    dialog = themed_app.get_by_test_id("stModal")
+    expect(dialog).to_be_visible()
+    assert_snapshot(dialog, name="settings_dialog")
```

Comment:
nit: it was my suggestion to do the snapshot on `stModel`, but looking at it again.. i didn't expect that it would be a fullscreen snapshot (sorry, my fault). Fullscreen screenshots have a higher risk for flakiness and take more time to check and don't add a lot of value here. It's easy to fix here by just adding back the `get_by_role("dialog")` but calling it on dialog: 

```suggestion
    assert_snapshot(dialog.get_by_role("dialog"), name="settings_dialog")
```